### PR TITLE
Add UI test to ps banner module

### PR DESCRIPTION
**Pull request by boubkerbribri**
| Questions         | Answers
| ------------------| -------------------------------------------------------
| Description?      | Add UI test to ps banner module with a github action that run every pull request
| Type?             | refacto
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | no
| How to test?      | GA is good https://github.com/boubkerbribri/ps_banner/runs/1661558991?check_suite_focus=true
| Possible impacts? | no impacts

### CODE REVIEW

**Review by SimonGrn — APPROVED** [review]

File: tests/E2E/README.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+These tests can be run on `1.7.7.x` and `1.7.6.x` PrestaShop versions.
+
+## Stack
+
+In these tests, we used the actual stack :
+
+- [prestashop_test_lib](https://www.npmjs.com/package/prestashop_test_lib) which uses [playwright](https://playwright.dev/) as a node library to automate browsers.
+- [mocha](https://mochajs.org/) as a test framework.
+- [chai](https://www.chaijs.com/) as an assertion library.
+- [js-image-generator](https://www.npmjs.com/package/js-image-generator) to generate fake images for tests.
```
Comment (Progi1984) [inline tests/E2E/README.md:16]:
```suggestion
- [js-image-generator](https://www.npmjs.com/package/js-image-generator) as a node library to generate fake images for tests.
```

File: tests/E2E/README.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+## Running tests
+
+Before running tests, you should run `npm install`, to install all dependencies and to download browsers.
```
Comment (Progi1984) [inline tests/E2E/README.md:20]:
```suggestion
Before running tests, you should run `npm install`, to install all dependencies.
```

File: tests/E2E/README.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+### Running all tests
+
+```shell
+PS_VERSION='1.7.7' URL_FO='http://localhost.com/' npm run e2e-tests 
+```
+
+### Running all tests with fast fail mode
+
+To run test that stop after the first fail:
```
Comment (Progi1984) [inline tests/E2E/README.md:30]: explain fast fail mode

**Review by Progi1984 — COMMENTED** [review]
Small feedbacks

File: tests/E2E/README.md (hunk 3 above)
Comment (boubkerbribri) [reply tests/E2E/README.md]: done

File: tests/E2E/README.md (hunk 2 above)
Comment (SimonGrn) [reply tests/E2E/README.md]: It's downloading browsers if playwright is in the dependencies, and that represents a huge file... this warning wasn't too much I think.

Comment (Progi1984) [reply tests/E2E/README.md]: You're right @SimonGrn .

**Review by Progi1984 — DISMISSED** [review]

File: tests/E2E/README.md (hunk 2 above)
Comment (SimonGrn) [reply tests/E2E/README.md]: As usual, what else is new 😎

**Review by matks — APPROVED** [review]
@PierreRambaud If we merge this PR, upon next release all of this `.github` and `tests` folders will be put inside the ZIP folder. So these files will be exposed. Wdyt ?

**Comment by PierreRambaud** [comment]
> @PierreRambaud If we merge this PR, upon next release all of this `.github` and `tests` folders will be put inside the ZIP folder. So these files will be exposed. Wdyt ?

We have to be sure these files are not in the zip archive

**Comment by NeOMakinG** [comment]
> > @PierreRambaud If we merge this PR, upon next release all of this `.github` and `tests` folders will be put inside the ZIP folder. So these files will be exposed. Wdyt ?
> 
> We have to be sure these files are not in the zip archive

How can we manage to avoid this?

**Comment by PierreRambaud** [comment]
> > > @PierreRambaud If we merge this PR, upon next release all of this `.github` and `tests` folders will be put inside the ZIP folder. So these files will be exposed. Wdyt ?
> > 
> > 
> > We have to be sure these files are not in the zip archive
> 
> How can we manage to avoid this?

I will investigate tomorrow :sweat_smile:

**Comment by PierreRambaud** [comment]
FYI, regarding the code executed by the Addons platform, all directories are includes when it uses the https://github.com/PrestaShop/PrestaShop-modules repository.

**Review by Progi1984 — APPROVED** [review]

**Comment by Progi1984** [comment]
@PierreRambaud Ok for merge ot not ?

**Review by PierreRambaud — APPROVED** [review]

**Comment by PierreRambaud** [comment]
:tada: Thank you @boubkerbribri

**Pull request merged**